packages/sit-onyx/src/components/OnyxMenuItem/OnyxMenuItem.vue

```diff
@@ -0,0 +1,70 @@
+<script setup lang="ts">
+import { inject } from "vue";
+import OnyxListItem from "../OnyxListItem/OnyxListItem.vue";
+import { MENU_BUTTON_ITEM_INJECTION_KEY } from "../OnyxNavItem/future/types";
+import { type OnyxMenuItemProps } from "./types";
+
+const props = defineProps<OnyxMenuItemProps>();
+const emit = defineEmits<{
+  /**
+   * Emitted when the menu item is clicked (via click or keyboard).
+   */
+  click: [];
+}>();
+
+const menuButton = inject(MENU_BUTTON_ITEM_INJECTION_KEY);
+</script>
+<template>
+  <OnyxListItem
+    :selected="props.active"
+    v-bind="menuButton?.listItem"
+    :color="props.color"
+    :disabled="props.disabled"
+    class="onyx-menu-item"
+  >
+    <a
+      v-if="props.href"
+      class="onyx-future-nav-item__anchor"
+      v-bind="menuButton?.menuItem({ active: props.active })"
+      :href="props.href"
+      @click="emit('click')"
+    >
+      <slot></slot>
+    </a>
+    <button
+      v-else
+      class="onyx-menu-item__button onyx-list-item"
+      :class="{
+        'onyx-list-item--active': props.active,
+        [`onyx-list-item--${props.color}`]: props.color,
+        'onyx-list-item--disabled': props.disabled,
+      }"
+      v-bind="menuButton?.menuItem({ active: props.active })"
+      @click="emit('click')"
+    >
+      <slot></slot>
+    </button>
+  </OnyxListItem>
```

Comment:
We can group this logic together:
```suggestion
    <component
      is="props.href ? 'a' : 'button'"
      :class="{
        'onyx-menu-item': true,
         'onyx-menu-item__anchor': props.href,
         'onyx-menu-item__button': !props.href,
        'onyx-list-item--active': props.active,
        [`onyx-list-item--${props.color}`]: props.color,
        'onyx-list-item--disabled': props.disabled,
      }"
      v-bind="menuButton?.menuItem({ active: props.active })"
      :href="props.href"
      @click="emit('click')"
    >
      <slot></slot>
    </component>
  </OnyxListItem>
```